frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts

```diff
@@ -299,21 +329,28 @@ export class ConsoleRemotePlugin implements webpack.WebpackPluginInstance {
       validateConsoleProvidedSharedModules(this.pkg).report();
     }
 
+    const resolvedModulePaths = this.adaptedOptions.sharedDynamicModuleSettings.modulePaths ?? [
+      path.resolve(process.cwd(), 'node_modules'),
+    ];
+
     this.sharedDynamicModuleMaps = Object.entries(
       this.adaptedOptions.sharedDynamicModuleSettings.packageSpecs ?? {
         '@patternfly/react-core': {},
         '@patternfly/react-icons': {},
         '@patternfly/react-table': {},
       },
     ).reduce<Record<string, DynamicModuleMap>>(
-      (acc, [pkgName, { indexModule = 'dist/esm/index.js', resolutionField = 'module' }]) => ({
-        ...acc,
-        [pkgName]: getDynamicModuleMap(
-          path.resolve(this.baseDir, 'node_modules', pkgName),
-          indexModule,
-          resolutionField,
-        ),
-      }),
+      (acc, [pkgName, { indexModule = 'dist/esm/index.js', resolutionField = 'module' }]) => {
+        const basePath = resolvedModulePaths
+          .map((p) => path.resolve(p, pkgName))
+          .find((p) => fs.existsSync(p) && fs.statSync(p).isDirectory());
+
+        if (basePath) {
+          acc[pkgName] = getDynamicModuleMap(basePath, indexModule, resolutionField);
+        }
+
+        return acc;
```

Comment:
What if we used an immutable accumulator like this:
```suggestion
        return !basePath ? acc : { 
          ...acc, 
          [pkgName]: getDynamicModuleMap(basePath, indexModule, resolutionField)
         };
```

Reply:
@TheRealJon Good idea, I'd personally prefer non-negated variation like
```ts
return basePath
  ? { ...acc, [pkgName]: getDynamicModuleMap(basePath, indexModule, resolutionField) }
  : acc;
```